### PR TITLE
linux_xanmod: only apply fix for removed options on tt variant

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
Those options are available in the normal variants.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 186091` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>37 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_xanmod.ixgbevf</li>
    <li>linuxKernel.packages.linux_xanmod.mxu11x0</li>
    <li>linuxKernel.packages.linux_xanmod.ndiswrapper</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy340</li>
    <li>linuxKernel.packages.linux_xanmod.phc-intel</li>
    <li>linuxKernel.packages.linux_xanmod.r8125</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8723bs</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod.sch_cake</li>
    <li>linuxKernel.packages.linux_xanmod.tbs</li>
    <li>linuxKernel.packages.linux_xanmod_latest.akvcam</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ax99100</li>
    <li>linuxKernel.packages.linux_xanmod_latest.can-isotp</li>
    <li>linuxKernel.packages.linux_xanmod_latest.dpdk</li>
    <li>linuxKernel.packages.linux_xanmod_latest.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ena</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ixgbevf</li>
    <li>linuxKernel.packages.linux_xanmod_latest.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod_latest.mwprocapture</li>
    <li>linuxKernel.packages.linux_xanmod_latest.mxu11x0</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ndiswrapper</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy340</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_vulkan_beta</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidiabl</li>
    <li>linuxKernel.packages.linux_xanmod_latest.phc-intel</li>
    <li>linuxKernel.packages.linux_xanmod_latest.r8125</li>
    <li>linuxKernel.packages.linux_xanmod_latest.r8168</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8189es</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8192eu</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8723bs</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl88xxau-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod_latest.sch_cake</li>
    <li>linuxKernel.packages.linux_xanmod_latest.tbs</li>
  </ul>
</details>
<details>
  <summary>150 packages built:</summary>
  <ul>
    <li>linuxKernel.kernels.linux_xanmod</li>
    <li>linuxKernel.kernels.linux_xanmod_latest</li>
    <li>linuxKernel.packages.linux_xanmod.acpi_call</li>
    <li>linuxKernel.packages.linux_xanmod.akvcam</li>
    <li>linuxKernel.packages.linux_xanmod.apfs</li>
    <li>linuxKernel.packages.linux_xanmod.asus-ec-sensors</li>
    <li>linuxKernel.packages.linux_xanmod.asus-wmi-sensors</li>
    <li>linuxKernel.packages.linux_xanmod.ax99100</li>
    <li>linuxKernel.packages.linux_xanmod.batman_adv</li>
    <li>linuxKernel.packages.linux_xanmod.bbswitch</li>
    <li>linuxKernel.packages.linux_xanmod.broadcom_sta</li>
    <li>linuxKernel.packages.linux_xanmod.can-isotp</li>
    <li>linuxKernel.packages.linux_xanmod.chipsec</li>
    <li>linuxKernel.packages.linux_xanmod.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod.ddcci-driver</li>
    <li>linuxKernel.packages.linux_xanmod.dddvb</li>
    <li>linuxKernel.packages.linux_xanmod.digimend</li>
    <li>linuxKernel.packages.linux_xanmod.dpdk</li>
    <li>linuxKernel.packages.linux_xanmod.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_xanmod.ena</li>
    <li>linuxKernel.packages.linux_xanmod.evdi</li>
    <li>linuxKernel.packages.linux_xanmod.facetimehd</li>
    <li>linuxKernel.packages.linux_xanmod.fwts-efi-runtime</li>
    <li>linuxKernel.packages.linux_xanmod.gcadapter-oc-kmod</li>
    <li>linuxKernel.packages.linux_xanmod.hid-ite8291r3</li>
    <li>linuxKernel.packages.linux_xanmod.hid-nintendo</li>
    <li>linuxKernel.packages.linux_xanmod.isgx</li>
    <li>linuxKernel.packages.linux_xanmod.it87</li>
    <li>linuxKernel.packages.linux_xanmod.ithc</li>
    <li>linuxKernel.packages.linux_xanmod.jool</li>
    <li>linuxKernel.packages.linux_xanmod.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod.kvmfr</li>
    <li>linuxKernel.packages.linux_xanmod.liquidtux</li>
    <li>linuxKernel.packages.linux_xanmod.lttng-modules</li>
    <li>linuxKernel.packages.linux_xanmod.mba6x_bl</li>
    <li>linuxKernel.packages.linux_xanmod.mbp2018-bridge-drv</li>
    <li>linuxKernel.packages.linux_xanmod.mwprocapture</li>
    <li>linuxKernel.packages.linux_xanmod.netatop</li>
    <li>linuxKernel.packages.linux_xanmod.new-lg4ff</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_beta</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_vulkan_beta</li>
    <li>linuxKernel.packages.linux_xanmod.nvidiabl</li>
    <li>linuxKernel.packages.linux_xanmod.openafs</li>
    <li>linuxKernel.packages.linux_xanmod.openrazer</li>
    <li>linuxKernel.packages.linux_xanmod.prl-tools</li>
    <li>linuxKernel.packages.linux_xanmod.qc71_laptop</li>
    <li>linuxKernel.packages.linux_xanmod.r8168</li>
    <li>linuxKernel.packages.linux_xanmod.rr-zen_workaround</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8189es</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8189fs</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8192eu</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8814au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8821au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8821ce</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8821cu</li>
    <li>linuxKernel.packages.linux_xanmod.rtl88x2bu</li>
    <li>linuxKernel.packages.linux_xanmod.rtl88xxau-aircrack</li>
    <li>linuxKernel.packages.linux_xanmod.rtlwifi_new</li>
    <li>linuxKernel.packages.linux_xanmod.rtw89</li>
    <li>linuxKernel.packages.linux_xanmod.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod.system76</li>
    <li>linuxKernel.packages.linux_xanmod.system76-acpi</li>
    <li>linuxKernel.packages.linux_xanmod.system76-io</li>
    <li>linuxKernel.packages.linux_xanmod.systemtap</li>
    <li>linuxKernel.packages.linux_xanmod.tp_smapi</li>
    <li>linuxKernel.packages.linux_xanmod.tuxedo-keyboard</li>
    <li>linuxKernel.packages.linux_xanmod.v4l2loopback</li>
    <li>linuxKernel.packages.linux_xanmod.v86d</li>
    <li>linuxKernel.packages.linux_xanmod.veikk-linux-driver</li>
    <li>linuxKernel.packages.linux_xanmod.vendor-reset</li>
    <li>linuxKernel.packages.linux_xanmod.vhba</li>
    <li>linuxKernel.packages.linux_xanmod.virtio_vmmci</li>
    <li>linuxKernel.packages.linux_xanmod.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_xanmod.vmm_clock</li>
    <li>linuxKernel.packages.linux_xanmod.vmware</li>
    <li>linuxKernel.packages.linux_xanmod.xone</li>
    <li>linuxKernel.packages.linux_xanmod.xpadneo</li>
    <li>linuxKernel.packages.linux_xanmod.zenpower</li>
    <li>linuxKernel.packages.linux_xanmod.zfs (linuxKernel.packages.linux_xanmod.zfsUnstable)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.acpi_call</li>
    <li>linuxKernel.packages.linux_xanmod_latest.apfs</li>
    <li>linuxKernel.packages.linux_xanmod_latest.asus-ec-sensors</li>
    <li>linuxKernel.packages.linux_xanmod_latest.asus-wmi-sensors</li>
    <li>linuxKernel.packages.linux_xanmod_latest.batman_adv</li>
    <li>linuxKernel.packages.linux_xanmod_latest.bbswitch</li>
    <li>linuxKernel.packages.linux_xanmod_latest.broadcom_sta</li>
    <li>linuxKernel.packages.linux_xanmod_latest.chipsec</li>
    <li>linuxKernel.packages.linux_xanmod_latest.cryptodev</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ddcci-driver</li>
    <li>linuxKernel.packages.linux_xanmod_latest.dddvb</li>
    <li>linuxKernel.packages.linux_xanmod_latest.digimend</li>
    <li>linuxKernel.packages.linux_xanmod_latest.evdi</li>
    <li>linuxKernel.packages.linux_xanmod_latest.facetimehd</li>
    <li>linuxKernel.packages.linux_xanmod_latest.fwts-efi-runtime</li>
    <li>linuxKernel.packages.linux_xanmod_latest.gcadapter-oc-kmod</li>
    <li>linuxKernel.packages.linux_xanmod_latest.hid-ite8291r3</li>
    <li>linuxKernel.packages.linux_xanmod_latest.hid-nintendo</li>
    <li>linuxKernel.packages.linux_xanmod_latest.isgx</li>
    <li>linuxKernel.packages.linux_xanmod_latest.it87</li>
    <li>linuxKernel.packages.linux_xanmod_latest.ithc</li>
    <li>linuxKernel.packages.linux_xanmod_latest.jool</li>
    <li>linuxKernel.packages.linux_xanmod_latest.kvmfr</li>
    <li>linuxKernel.packages.linux_xanmod_latest.liquidtux</li>
    <li>linuxKernel.packages.linux_xanmod_latest.lttng-modules</li>
    <li>linuxKernel.packages.linux_xanmod_latest.mba6x_bl</li>
    <li>linuxKernel.packages.linux_xanmod_latest.mbp2018-bridge-drv</li>
    <li>linuxKernel.packages.linux_xanmod_latest.netatop</li>
    <li>linuxKernel.packages.linux_xanmod_latest.new-lg4ff</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_beta</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_beta_open</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_xanmod_latest.openafs</li>
    <li>linuxKernel.packages.linux_xanmod_latest.openrazer</li>
    <li>linuxKernel.packages.linux_xanmod_latest.prl-tools</li>
    <li>linuxKernel.packages.linux_xanmod_latest.qc71_laptop</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rr-zen_workaround</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8189fs</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8814au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8821au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8821ce</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8821cu</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl88x2bu</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtlwifi_new</li>
    <li>linuxKernel.packages.linux_xanmod_latest.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod_latest.system76</li>
    <li>linuxKernel.packages.linux_xanmod_latest.system76-acpi</li>
    <li>linuxKernel.packages.linux_xanmod_latest.system76-io</li>
    <li>linuxKernel.packages.linux_xanmod_latest.systemtap</li>
    <li>linuxKernel.packages.linux_xanmod_latest.tp_smapi</li>
    <li>linuxKernel.packages.linux_xanmod_latest.tuxedo-keyboard</li>
    <li>linuxKernel.packages.linux_xanmod_latest.v4l2loopback</li>
    <li>linuxKernel.packages.linux_xanmod_latest.v86d</li>
    <li>linuxKernel.packages.linux_xanmod_latest.veikk-linux-driver</li>
    <li>linuxKernel.packages.linux_xanmod_latest.vendor-reset</li>
    <li>linuxKernel.packages.linux_xanmod_latest.vhba</li>
    <li>linuxKernel.packages.linux_xanmod_latest.virtio_vmmci</li>
    <li>linuxKernel.packages.linux_xanmod_latest.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod_latest.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_xanmod_latest.vmm_clock</li>
    <li>linuxKernel.packages.linux_xanmod_latest.vmware</li>
    <li>linuxKernel.packages.linux_xanmod_latest.xone</li>
    <li>linuxKernel.packages.linux_xanmod_latest.xpadneo</li>
    <li>linuxKernel.packages.linux_xanmod_latest.zenpower</li>
    <li>linuxKernel.packages.linux_xanmod_latest.zfs (linuxKernel.packages.linux_xanmod_latest.zfsUnstable)</li>
  </ul>
</details>

**Review by moni-dz — APPROVED** [review]

**Review by Atemu — APPROVED** [review]

**Comment by QuantMint** [comment]
I am pretty sure that the same can be done for:
SCHED_CORE
PREEMPT_VOLUNTARY
PREEMPT
NO_HZ_FULL
They either have the opposite value or they are not present at all in:
https://github.com/xanmod/linux/blob/5.18/CONFIGS/xanmod/gcc/config_x86-64-v2

**Comment by NickCao** [comment]
> I am pretty sure that the same can be done for: SCHED_CORE PREEMPT_VOLUNTARY PREEMPT NO_HZ_FULL They either have the opposite value or they are not present at all in: https://github.com/xanmod/linux/blob/5.18/CONFIGS/xanmod/gcc/config_x86-64

Futher chaked https://github.com/xanmod/linux/blob/5.15-tt/CONFIGS/xanmod/gcc/config and https://github.com/xanmod/linux/blob/5.15/CONFIGS/xanmod/gcc/config, looks like they can just be removed.

**Comment by NickCao** [comment]
May I ask how can we generate an up to date list of `structuredExtraConfig`, the current approach of manually comparing the differences is a bit cumbersome.

File: pkgs/os-specific/linux/kernel/xanmod-kernels.nix [hunk 1]
```diff
@@ -42,13 +40,6 @@ let
       NTFS3_LZX_XPRESS = yes;
       NTFS3_FS_POSIX_ACL = yes;
 
-      # Preemptive Full Tickless Kernel at 500Hz
-      SCHED_CORE = lib.mkForce (option no);
-      PREEMPT_VOLUNTARY = lib.mkForce no;
-      PREEMPT = lib.mkForce yes;
```
Comment (fortuneteller2k) [inline pkgs/os-specific/linux/kernel/xanmod-kernels.nix:48]: The kernel common config would produce the opposite result with `PREEMPT_*` disabled.

**Review by moni-dz — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/kernel/xanmod-kernels.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/os-specific/linux/kernel/xanmod-kernels.nix:48]: But `PREEMPT_VOLUNTARY` is set to y in upstream xanmod config,

Comment (fortuneteller2k) [reply pkgs/os-specific/linux/kernel/xanmod-kernels.nix:48]: Ah, I see.

**Review by moni-dz — APPROVED** [review]

**Review by Tungsten842 — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Can you fix the conflicts?

**Comment by ElXreno** [comment]
In my opinion, `HZ_500` should only be removed for the `tt` variant, the other variants still use it.
References: [tt variant](https://github.com/xanmod/linux/blob/5.15-tt/CONFIGS/xanmod/gcc/config#L510) | [stable variant](https://github.com/xanmod/linux/blob/5.18/CONFIGS/xanmod/gcc/config_x86-64#L521) | [edge variant](https://github.com/xanmod/linux/blob/5.19/CONFIGS/xanmod/gcc/config_x86-64#L513)

**Comment by NickCao** [comment]
`HZ_500` should be a nop, as `HZ` is explicitly set for these kernels, see https://github.com/torvalds/linux/search?q=HZ_500

**Pull request merged**